frontend/src/views/Lobby.tsx

```diff
@@ -143,6 +143,7 @@ function LobbyPage() {
   const [difficulty, setDifficulty] = useState<Difficulty | null>(null);
   const [duration, setDuration] = useState<number | undefined>(15);
   const [size, setSize] = useState<number | undefined>(10);
+  const [numProblems, setNumProblems] = useState<number | undefined>(1);
```

Comment:
nit: this can be simply of type `<number>` (or no type at all, since it can infer that it's a number from the default value)